src/Library/Plugin/Plugin.php

```diff
@@ -113,25 +120,40 @@ class Plugin {
 	 */
 	protected $slug;
 
+	/**
+	 * Plugin Update Data.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @var Response
+	 */
+	public $updateData;
+
 	/**
 	 * Constructor.
 	 *
 	 * @since 2.7.7
 	 *
-	 * @param string $slug For example: "plugin" from plugin/plugin.php
+	 * @param string $slug For example: "plugin" from plugin/plugin.php.
+	 * @param array  $pluginConfig An array of plugin config data.
 	 */
-	public function __construct( $slug, $pluginConfig = null ) {
-		$this->slug = $slug;
+	public function __construct( $slug = null, $pluginConfig = null, $file = null ) {
```

Comment:
yes. hello.php is one of them.

Reply:
This is starting to get confusing. How do I create a new plugin? Do I pass in a slug, a file, a slug and a file?

What do you think about creating a factory such as **src/Library/Plugin/Factory.php** that will get a plugin for us? It could be something like this:

```
class Factory {
    public function getBySlug( $slug )

    public function getByFile( $file )
}
```